### PR TITLE
change placement group report size to 1k

### PR DESCRIPTION
**Pull request by jiaodong**
## Why are these changes needed?

We're adding nightly tests for ray serve at 1k replica. We noticed that it took ~22 mins for autoscaler to reach 1k replicas, from auto scaler log it shows we're adding replicas in a batch of 100. If there's no other concerns we can raise this config to 1000 so autoscaler can get 1 placement group request and process faster.

cc: @simon-mo 

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by ericl — APPROVED** [review]

**Comment by ericl** [comment]
LGTM but fyi @AmeerHajAli

**Review by ijrsvt — APPROVED** [review]
This LGTM to me

OOC are these values overridable via environment variables? 

Also, do we warn anywhere if there are more placement_groups than this value?

**Comment by AmeerHajAli** [comment]
I think this change should be OK.
Cc @mwtian @DmitriGekhtman

**Comment by rkooo567** [comment]
Can you guys run some workload to verify this won't introduce any issue after merging it?

**Review by DmitriGekhtman — APPROVED** [review]

**Comment by wuisawesome** [comment]
I think this is fine to merge (once tests pass). I think there are also some tune users suffering from slow autoscaling due to this.

**Review by wuisawesome — APPROVED** [review]

**Comment by jiaodong** [comment]
Thanks for the quick response guys ! I can update in this PR with a new test run using 1k replica and compare the time needed to spawn them in autoscaler.

In the meantime i'm not committer of ray repo yet so i cannot merge this myself .. just fyi ;)

**Comment by wuisawesome** [comment]
When the CI finishes, just replace the "author action required" label with "test ok" and someone will come along and merge it. (everyone assigned is a committer).

**Comment by jiaodong** [comment]
I launched a new job on ```test_wheels/larger_placement_group``` branch with this change but didn't notice difference in autoscale behavior or reduced starting up time (~21min) , did I miss anything ?

```
 {'CPU': 1.0} * 1 (PACK): 100+ pending placement groups
2021-07-20 13:27:37,454	INFO monitor.py:224 -- :event_summary:Adding 13 nodes of type worker_node.
2021-07-20 13:27:38,911	INFO node_launcher.py:96 -- NodeLauncher0: Got 3 nodes to launch.
2021-07-20 13:27:38,940	INFO node_launcher.py:96 -- NodeLauncher0: Launching 3 nodes, type worker_node.
2021-07-20 13:27:38,977	INFO node_provider.py:438 -- Configuration bypasses Rapid Start; launching node normally.
2021-07-20 13:27:45,027	WARNING resource_demand_scheduler.py:713 -- The autoscaler could not find a node type to satisfy the request: [{'CPU_group_8b5e7ffec279dbf65a04312bb52d6d1f': 1.0}, {'CPU_group_85dd1c480c7cf90474d630d47e3d917c': 1.0},
```

**Comment by ijrsvt** [comment]
@wuisawesome Just to check pending placement_groups (beyond the **max_report_size**)  will eventually get put into GCS, but just after placement_groups in there are scheduled?

**Comment by wuisawesome** [comment]
I believe GCS should always store all placement groups. This param is purely about how many we send the autoscaler.

**Comment by rkooo567** [comment]
What's the status of this PR?

**Comment by jiaodong** [comment]
> What's the status of this PR?

I didn't see significant change of cluster starting up time to 1k replica but it should be mostly due to min_worker=0 , max_worker=130 and it only scales by incremental small batches. Changing min_worker=130 is much more effective.

So for my particular issue this PR didn't make a huge difference, but since we got multiple approvals on this change, it might not hurt to tune it up ? cc: @iycheng who I saw also working with large # of ray actors through autoscaler recently

**Comment by ericl** [comment]
Let's see what happens then.

**Pull request merged**
